actionSHARK/github.py

```diff
@@ -169,6 +170,11 @@ def paginating(self, github_url: str, checker: Optional[str] = None):
             # count number of items
             response_count = len(response_JSON)
 
+            if self.current_action == 'run':
```

Comment:
I see why you put this in there. The paginating() method could also be called paginate_rate_limit_save_to_db.
While I see the case for the rate_limiting code I do not see it for directly saving to the DB in this function.
I think the run_ids should come from a result of the get_runs method.

Could you refeactor the paginate method to actually return results which are then saved in their respective methods, e.g., get_runs, get_workflows? 
Now that I am looking at it I am also not happy with the whole DB adapter code in mongo.py but we will leave that for now if we can.